fibonacci/julia/code.jl

```diff
@@ -1,15 +1,5 @@
-# base case 1
-fibonacci(::Val{0}) = 0
-# base case 2
-fibonacci(::Val{1}) = 1
-# general case
-fibonacci(::Val{n}) where n = fibonacci(Val(n-1)) + fibonacci(Val(n-2))
+fibonacci(n) = n < 2 ? n : fibonacci(n-1) + fibonacci(n-2)
 
-let
-    u = parse(Int,ARGS[1])
-    r = 0
-    for i ∈ 1:u
-        r += fibonacci(Val(i))
-    end
-    println(r)
-end
\ No newline at end of file
+main(u) = println(sum(fibonacci, 1:u))
```

Comment:
Can you explain if / how this is equivalent to the loop?

Reply:
The `sum` function just calls `mapreduce` with addition as the reducing function. So this is equivalent to mapping `fibonacci` to all elements in the range from 1 to `u` and then adding them up.

If you're prioritizing making the implementations more uniform between languages rather than using more idiomatic language constructs, going back to a loop is just fine. Let me know.